evennia/contrib/mail.py

```diff
@@ -0,0 +1,220 @@
+"""
+In-Game Mail system
+
+Evennia Contribution - grungies1138 2016
+
+A simple Brandymail style @mail system that uses the Msg class from Evennia Core.
+
+Installation:
+    import MailCommand from this module into the default Player or Character command set
+"""
+
+from evennia import default_cmds, search_object
+from evennia.utils import create, utils, evtable, evform
+from evennia.comms.models import Msg
+
+
+_HEAD_CHAR = "|015-|n"
+_SUB_HEAD_CHAR = "-"
+_WIDTH = 78
+
+class CmdMail(default_cmds.MuxCommand):
+    """
+    Commands that allow either IC or OOC communications
+
+    Usage:    
+        {w@mail{n
+                - Displays all the mail a player has in their mailbox
+
+        {w@mail <#>{n
+                - Displays a specific message
+
+        {w@mail <players>=<subject>/<message>{n
+                - Sends a message to the comma separated list of players.
+
+        {w@mail/delete <#>{n
+                - Deletes a specific message
+
+        {w@mail/forward <player list>=<#>[/<Message>]{n
+                - Forwards an existing message to the specified list of players,
+                  original message is delivered with optional Message prepended.
+
+        {w@mail/reply <#>=<message>{n
+                - Replies to a message #.  Prepends message to the original
+                  message text.
+    Switches:
+        delete  - deletes a message
+        forward - forward a received message to another object with an optional message attached.
+        reply   - Replies to a received message, appending the original message to the bottom.
+
+    Status Legend:
+        '|wU|n' = Unread/New message
+        '|wO|n' = Read/Old message
+        '|wR|n' = Replied to
+        '|wF|n' = Forwarded
+        
+    Examples:
+        @mail 2
+        @mail Griatch=New mail/Hey man, I am sending you a message!
+        @mail/delete 6
+        @mail/forward feend78 Griatch=You guys should read this.
+        @mail/reply 9=Thanks for the info!
+    """
+    key = "@mail"
+    aliases = ["mail"]
+    lock = "cmd:all()"
+    help_category = "General"
+
+    def func(self):
+        subject = ""
+        body = ""
+        if self.switches or self.args:
+            if "delete" in self.switches:
+                if not self.lhs:
+                    self.caller.msg("No Message ID given.  Unable to delete.")
+                    return
+                else:
+                    if self.get_all_mail()[int(self.lhs) - 1]:
```

Comment:
All the cases where you do int(self.lhs) is susceptible to a traceback if the user enters something other than a number. For example `@mail blah` currently gives a traceback. There are many such crashes in the code below due to invalidated input.